credentials/google/xds.go

```diff
@@ -50,18 +53,36 @@ func newClusterTransportCreds(tls, alts credentials.TransportCredentials) *clust
 	}
 }
 
-func (c *clusterTransportCreds) ClientHandshake(ctx context.Context, authority string, rawConn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+func isXDSNonCFECluster(ctx context.Context) bool {
 	chi := credentials.ClientHandshakeInfoFromContext(ctx)
 	if chi.Attributes == nil {
-		return c.tls.ClientHandshake(ctx, authority, rawConn)
+		return false
 	}
 	cn, ok := internal.GetXDSHandshakeClusterName(chi.Attributes)
 	if !ok || strings.HasPrefix(cn, cfeClusterNamePrefix) {
```

Comment:
Do we need the second half of this conditional statement?

Reply:
OH, sorry I see it is repeated below.  I think that's because I combined C's code with what we already had.  I will delete the duplicate `if HasPrefix` below.